flink-table/flink-table-planner/src/main/java/org/apache/calcite/rel/rules/MultiJoin.java

```diff
@@ -0,0 +1,306 @@
+        }
+        return clonedMap;
+    }
+
+    @Override
+    public RelWriter explainTerms(RelWriter pw) {
+        List<String> joinTypeNames = new ArrayList<>();
+        List<String> outerJoinConds = new ArrayList<>();
+        List<String> projFieldObjects = new ArrayList<>();
+        for (int i = 0; i < inputs.size(); i++) {
+            joinTypeNames.add(joinTypes.get(i).name());
+            RexNode outerJoinCondition = outerJoinConditions.get(i);
+            if (outerJoinCondition == null) {
+                outerJoinConds.add("NULL");
+            } else {
+                outerJoinConds.add(outerJoinCondition.toString());
+            }
+            ImmutableBitSet projField = projFields.get(i);
+            if (projField == null) {
+                projFieldObjects.add("ALL");
+            } else {
+                projFieldObjects.add(projField.toString());
+            }
+        }
+
+        super.explainTerms(pw);
+        for (Ord<RelNode> ord : Ord.zip(inputs)) {
+            pw.input("input#" + ord.i, ord.e);
+        }
+        return pw.item("joinFilter", joinFilter)
+                .item("isFullOuterJoin", isFullOuterJoin)
+                .item("joinTypes", joinTypeNames)
+                .item("outerJoinConditions", outerJoinConds)
+                .item("projFields", projFieldObjects)
+                .itemIf("postJoinFilter", postJoinFilter, postJoinFilter != null)
+                .itemIf("hints", hints, !hints.isEmpty());
+    }
+
+    @Override
+    public RelDataType deriveRowType() {
+        return rowType;
+    }
+
+    @Override
+    public List<RelNode> getInputs() {
+        return inputs;
+    }
+
+    @Override
+    public RelNode accept(RexShuttle shuttle) {
+        RexNode joinFilter = shuttle.apply(this.joinFilter);
+        List<@Nullable RexNode> outerJoinConditions = shuttle.apply(this.outerJoinConditions);
+        RexNode postJoinFilter = shuttle.apply(this.postJoinFilter);
+
+        if (joinFilter == this.joinFilter
+                && outerJoinConditions == this.outerJoinConditions
+                && postJoinFilter == this.postJoinFilter) {
+            return this;
+        }
+
+        return new MultiJoin(
+                getCluster(),
+                hints,
+                inputs,
+                joinFilter,
+                rowType,
+                isFullOuterJoin,
+                outerJoinConditions,
+                joinTypes,
+                projFields,
+                joinFieldRefCountsMap,
+                postJoinFilter);
+    }
+
+    /** Returns join filters associated with this MultiJoin. */
+    public RexNode getJoinFilter() {
+        return joinFilter;
+    }
+
+    /** Returns true if the MultiJoin corresponds to a full outer join. */
+    public boolean isFullOuterJoin() {
+        return isFullOuterJoin;
+    }
+
+    /** Returns outer join conditions for null-generating inputs. */
```

Comment:
The comment seems to mismatch the method. The method just returns the outer join conditions or null if there are not any. What does this have to do with  `null-generating inputs.`